packages/program/src/contract-call-script.ts

```diff
@@ -204,6 +216,129 @@ const getFunctionOutputInfos = (functionScopes: InvocationScopeLike[]): CallOutp
     };
   });
 
+/**
+ * Obtains script data for a contract call according to the V0 specification.
+ *
+ * @param call - the contract call to obtain for script data for.
+ * @param segmentOffset - the segment to generate pointers and offset data from.
+ * @returns the populated script data and call parameter offsets.
+ */
+export const getScriptDataV0 = (
+  call: ContractCall,
+  segmentOffset: number
+): { scriptData: Uint8Array[]; callParamOffsets: CallOpcodeParamsOffset } => {
+  const scriptData: Uint8Array[] = [];
+  let gasForwardedSize: number = 0;
+
+  const callParamOffsets: CallOpcodeParamsOffset = {
+    amountOffset: segmentOffset,
+    assetIdOffset: segmentOffset + WORD_SIZE,
+    gasForwardedOffset: call.gas ? segmentOffset + WORD_SIZE + ASSET_ID_LEN : 0,
+    callDataOffset: segmentOffset + WORD_SIZE + ASSET_ID_LEN + gasForwardedSize,
+  };
+
+  /// script data, consisting of the following items in the given order:
+  /// 1. Amount to be forwarded `(1 * `[`WORD_SIZE`]`)`
```

Comment:
Is there any particular reason for the triple slashes and all the backticks?

Reply:
No actually sorry this is how they were previously. Regardless I'll change them.

Reply:
https://github.com/FuelLabs/fuels-ts/pull/1832/commits/f6a4658b7ccb8811e4aa2f9db899b20ccce6c6c2